static/js/templates.js

```diff
@@ -52,8 +54,19 @@ const processSearch = () => {
   const templates = document.querySelectorAll('.template-card')
   templates.forEach(
     elem =>
-      (elem.style = `display: ${!results.find(result => result.id === elem.id) ? 'none' : ''}`),
+      (elem.style = `display: ${
+        !results.find(result => result.id === elem.id) ? 'none' : ''
+      }`),
   )
+  // Remove section headers that contain no results
+  const sectionHeaders = document.querySelectorAll('#results>h2')
+  sectionHeaders.forEach(header => {
+    // all headers' next sibling is a `section` that parents the templates
+    let matches = [...header.nextElementSibling.childNodes].filter(el =>
+      results.find(result => result.id === el.id),
+    )
+    if (!matches.length) header.style.display = 'none'
```

Comment:
Does this re-add the header if a result is found in that category, after it's been hidden the first time?